src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,578 @@
+finished yet, a regular network fetch will happen.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/small-file.mp4">
+
+<video id="video" controls></video>
+
+<script>
+  // Later on, after some condition has been met, set video source to the
+  // preloaded video URL.
+  video.src = 'https://cdn.com/small-file.mp4';
+  video.play().then(_ => {
+    // If preloaded video URL was already cached, playback started immediately.
+  });
+</script>
+```
+
+Note: I would recommend using this only for audio files or short video clips (< 5MB).
+
+Because the preloaded resource is going to be consumed by a video element in
+the example, the `as` preload link value is `video`. If it were an audio
+element, it would be `as="audio"`.
+
+### Preload first segment
+
+The example below shows how to preload the first segment of a video with `<link
+rel="preload">` and use it with Media Source Extensions. If you're not familiar
+with the MSE Javascript API, please read [MSE basics].
+
+For the sake of simplicity, let's assume the entire video has been split into
+smaller files like "file_1.webm", "file_2.webm", "file_3.webm", etc. We'll use
+MSE to feed the video element with those once the `onload` event from `<link
+rel="preload">` has been fired.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/file_1.webm"
+    onload="preloadFinished()" onerror="preloadFailed()">
+
+<video id="video" controls></video>
+
+<script>
+  // This function is automatically executed when the first segment of video is
+  // preloaded successfully. We may call it explicitly as well later on.
+  function preloadFinished() {
+    const mediaSource = new MediaSource();
+    video.src = URL.createObjectURL(mediaSource);
+    mediaSource.addEventListener('sourceopen', sourceOpen);
+  }
+
+  function sourceOpen(event) {
+    URL.revokeObjectURL(video.src);
+    const mediaSource = event.target;
+    const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
+
+    // If video is preloaded already, fetch will return immediately a response
+    // from the browser cache (memory cache). Otherwise, it will perform a
+    // regular network fetch.
+    fetch('https://cdn.com/file_1.webm')
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      // Append the data into the new sourceBuffer.
+      sourceBuffer.appendBuffer(data);
+      // TODO: Fetch file_2.webm when user starts playing video.
+    })
+    .catch(error => {
+      // TODO: Show "Video is not available" message to user.
+    });
+  }
+
+  // This function is called if the first segment of video is not available.
+  function preloadFailed() {
+    // TODO: Show "Video is not available" message to user.
+  }
+</script>
+```
+
+Warning: For cross-origin resources, make sure your CORS headers are set
+properly because opaque responses retrieved with `fetch(videoFileUrl, { mode:
+'no-cors' })` are not allowed to feed any video or audio element.
```

Comment:
You can use opaques responses to video elements, in fact it's the default way media elements request things.

You need CORS to make MSE work, but it requires array buffers, and you can't create an array buffer from an opaque response.